lib/parsers.js

```diff
@@ -287,15 +287,30 @@ exports.parseColor = function parseColor(val) {
     alpha = 1;
   var parts;
   var res = colorRegEx1.exec(val);
-  // is it #aaa or #ababab
+  // is it #aaa, #ababab, #aaaa, #abababaa
   if (res) {
+    var defaultHex = val.substr(1);
     var hex = val.substr(1);
-    if (hex.length === 3) {
+    if ([3, 4].indexOf(defaultHex.length) > -1) {
       hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2];
+
+      if (defaultHex.length === 4) {
+        hex = hex + defaultHex[3] + defaultHex[3];
+      }
     }
     red = parseInt(hex.substr(0, 2), 16);
     green = parseInt(hex.substr(2, 2), 16);
     blue = parseInt(hex.substr(4, 2), 16);
+    if (hex.length === 8) {
+      var hexAlpha = hex.substr(6, 2);
+      var hexAlphaToRgbaAlpha = (parseInt(hexAlpha, 16) / 255).toFixed(2);
```

Comment:
Just wrapping this in one more `parseInt` would be abit more straightforward than adding a condition below:

```
var hexAlphaToRgbaAlpha = parseInt(parseInt(hexAlpha, 16) / 255).toFixed(2));
```

or even something like this:

```
var hexAlphaToRgbaAlphaUnrounded = parseInt(hexAlpha, 16) / 255);
var hexAlphaToRgbaAlpha = parseInt(hexAlphaToRgbaAlphaUnrounded.toFixed(2));
```

Reply:
But in doing so, if the alpha is between 0 and 1, like 0.67, it will be rounded and should not. I make this encoding below just in case it's 0 or 1, so I'm going to remove the decimal places.

Reply:
@GabrielBibiano based on the w3 spec and testing in Chrome / FF, I think it should be rounded to the nearest 1000th. So use `toFixed(3)`.

Using the `Number` constructor seems to remove unwanted decimal places (e.g. converts 0.500 to 0.5) which I think is what is intended by the second conditional being discussed. So we can use something like:

```
var hexAlphaToRgbaAlpha = Number((parseInt(hexAlpha, 16) / 255).toFixed(3))`
```

Reply:
Right! I'll make this modification.